backend/src/test/java/org/cryptomator/hub/license/LicenseHolderTest.java

```diff
@@ -157,6 +170,160 @@ public void testSetInvalidToken() {
+			Mockito.verify(validator, Mockito.times(2)).validate("token", "42");
+			Mockito.verify(session, Mockito.times(2)).persist(Mockito.eq(settingsMock));
+			Assertions.assertNull(holder.get());
+		}
+
+		@Test
+		@DisplayName("If token is set in DB, ignore valid init token")
+		public void testValidDBTokenIgnoresValidInitToken() {
+			var decodedJWT = Mockito.mock(DecodedJWT.class);
+			Mockito.when(validator.validate("token3000", "3000")).thenReturn(decodedJWT);
+			Settings settingsMock = new Settings();
+			settingsMock.hubId = "3000";
+			settingsMock.licenseKey = "token3000";
+			settingsClass.when(Settings::get).thenReturn(settingsMock);
+
+			holder.init();
+
+			// init implicitly called due to @PostConstruct which increases the times to verify by 1
+			// See https://github.com/cryptomator/hub/pull/229#discussion_r1374694626 for further information
+			Mockito.verify(validator, Mockito.times(2)).validate("token3000", "3000");
+			Assertions.assertEquals(decodedJWT, holder.get());
+		}
+
+		@Test
+		@DisplayName("If token is set in DB, ignore invalid init token")
+		public void testValidDBTokenIgnoresInvalidInitToken() {
+			Mockito.when(validator.validate("token", "42")).thenAnswer(invocationOnMock -> {
+				throw new JWTVerificationException("");
+			});
+
+			var decodedJWT = Mockito.mock(DecodedJWT.class);
+			Mockito.when(validator.validate("token3000", "3000")).thenReturn(decodedJWT);
+			Settings settingsMock = new Settings();
+			settingsMock.hubId = "3000";
+			settingsMock.licenseKey = "token3000";
+			settingsClass.when(Settings::get).thenReturn(settingsMock);
+
+			holder.init();
+
+			// init implicitly called due to @PostConstruct which increases the times to verify by 1
+			// See https://github.com/cryptomator/hub/pull/229#discussion_r1374694626 for further information
+			Mockito.verify(validator, Mockito.times(2)).validate("token3000", "3000");
+			Assertions.assertEquals(decodedJWT, holder.get());
+		}
+
+		@Test
+		@DisplayName("Setting a valid token validates and overwrites the init token")
+		public void testSetValidToken() {
+			var decodedJWT = Mockito.mock(DecodedJWT.class);
+			Mockito.when(validator.validate("token3000", "42")).thenReturn(decodedJWT);
+
+			Settings initSettingsMock = new Settings();
+			initSettingsMock.hubId = "42";
+			settingsClass.when(Settings::get).thenReturn(initSettingsMock);
+
+			Settings persistingSettingsMock = new Settings();
+			persistingSettingsMock.hubId = "42";
+			persistingSettingsMock.licenseKey = "token3000";
+
+			holder.set("token3000");
+
+			Mockito.verify(validator, Mockito.times(1)).validate("token3000", "42");
+			Mockito.verify(session, Mockito.times(1)).persist(Mockito.eq(persistingSettingsMock));
+			Assertions.assertEquals(decodedJWT, holder.get());
+		}
+
+		@Test
+		@DisplayName("Setting an invalid token fails with exception and falls back to previous token")
+		public void testSetInvalidToken() {
```

Comment:
this test should not be part of the nested InitTokenTest, because the matter to test is independent of wether the init token is set or not. It tests the setToken() functionality.

Reply:
This test is also available in the `TestSetter` class https://github.com/cryptomator/hub/blob/ecb4e696df88c02d092eaca6c4cac522a6322100/backend/src/test/java/org/cryptomator/hub/license/LicenseHolderTest.java#L155-L170

Here I want to verify that it still works with init token set but as you say, it shouldn't make a difference for this method, so if you prefer, I can delete it.

Reply:
Then remove it.

Reply:
Removed in [ff4aade](https://github.com/cryptomator/hub/pull/229/commits/ff4aaded404788c557b8ff892d5fd56a25b53653)